trinity/nodes/ropsten.py

```diff
@@ -11,7 +12,9 @@
 
 class RopstenFullNode(FullNode):
     chain_class = RopstenChain
+    initial_tx_validation_block_number = BYZANTIUM_ROPSTEN_BLOCK
 
 
 class RopstenLightNode(LightNode):
     chain_class = RopstenLightDispatchChain
+    initial_tx_validation_block_number = BYZANTIUM_ROPSTEN_BLOCK
```

Comment:
This solution is high maintenance: someone has to remember to update this variable in a bunch of different places at every fork. Or, realistically, no one does the maintenance. Then we get a totally silent bug: during syncing, we blacklists peers who send valid Constantinople transactions, if they are invalid in Byzantium.

One alternative is: we always drop transactions that fail validation based on our local node's canonical head. But we treat the peers differently during fast syncing and real-time sync...

- fast sync: do not drop or blacklist peers who send invalid transactions
- real-time sync: drop and blacklist peers who send invalid transactions

Reply:
>someone has to remember to update this variable in a bunch of different places at every fork

Yes, that's true. It's a bit better already in the PR that makes the transaction pool a plugin. It's more centralized there (hooray for centralization haha)

```python
    def activate(self, context: PluginContext) -> None:
        if isinstance(self.chain, MainnetChain):
            validator = DefaultTransactionValidator(self.chain, BYZANTIUM_MAINNET_BLOCK)
        elif isinstance(self.chain, RopstenChain):
            validator = DefaultTransactionValidator(self.chain, BYZANTIUM_ROPSTEN_BLOCK)
        else:
            # TODO: We could hint the user about e.g. a --tx-pool-no-validation flag to run the
            # tx pool without tx validation in this case
            raise ValueError("The TxPool plugin does only support MainnetChain or RopstenChain")

        tx_pool = TxPool(self.peer_pool, validator, self.cancel_token)
        asyncio.ensure_future(tx_pool.run())
```

We could probably even maintain some constant that points to whatever the *current fork* is at some central place such as `evm.constants`. It's not ideal but the alternatives aren't that compelling either imho. Given that forks happen rarely I feel it might be not a huge deal if can reduce it to one centrally maintained constant.